consensus/polybft/polybft_config.go

```diff
@@ -3,60 +3,91 @@ package polybft
 type PolyBFTConfig struct {
-	InitialValidatorSet []*Validator  `json:"initialValidatorSet"`
-	Bridge              *BridgeConfig `json:"bridge"`
+	// Validators are the genesis validators
+	Validators []*Validator `json:"validators"`
+
+	// Bridge is the rootchain bridge configuration
+	Bridge *BridgeConfig `json:"bridge"`
 
-	ValidatorSetSize int `json:"validatorSetSize"`
+	// ActiveValidatorSetSize denotes how many validators are active per each epoch
+	ActiveValidatorSetSize int `json:"activeValidatorSetSize"`
 
-	// Address of the system contracts, as of now (testing) this is populated automatically during genesis
-	ValidatorSetAddr  types.Address `json:"validatorSetAddr"`
-	StateReceiverAddr types.Address `json:"stateReceiverAddr"`
+	// EpochSize is size of epoch
+	EpochSize uint64 `json:"epochSize"`
 
-	// size of the epoch and sprint
-	EpochSize  uint64 `json:"epochSize"`
+	// SprintSize is size of sprint
 	SprintSize uint64 `json:"sprintSize"`
 
+	// BlockTime is target frequency of blocks production
 	BlockTime time.Duration `json:"blockTime"`
 
 	// Governance is the initial governance address
 	Governance types.Address `json:"governance"`
 }
 
+// GetPolyBFTConfig deserializes provided chain config and returns PolyBFTConfig
+func GetPolyBFTConfig(chainConfig *chain.Chain) (PolyBFTConfig, error) {
+	consensusConfigJSON, err := json.Marshal(chainConfig.Params.Engine[PolyBFTConsensusName])
+	if err != nil {
+		return PolyBFTConfig{}, err
+	}
+
+	var polyBFTConfig PolyBFTConfig
+	err = json.Unmarshal(consensusConfigJSON, &polyBFTConfig)
+
+	if err != nil {
+		return PolyBFTConfig{}, err
+	}
+
+	return polyBFTConfig, nil
+}
+
 // BridgeConfig is the rootchain bridge configuration
 type BridgeConfig struct {
-	BridgeAddr      types.Address `json:"stateSenderAddr"`
-	CheckpointAddr  types.Address `json:"checkpointAddr"`
-	JSONRPCEndpoint string        `json:"jsonRPCEndpoint"`
+	StateSenderAddress       types.Address `json:"stateSenderAddress"`
+	CheckpointManagerAddress types.Address `json:"checkpointManagerAddress"`
+	BLSAddress               types.Address `json:"blsAddress"`
+	BN256G2Address           types.Address `json:"bn256G2Address"`
+	AdminAddress             types.Address `json:"adminAddress"`
+	JSONRPCEndpoint          string        `json:"jsonRPCEndpoint"`
 }
 
 func (p *PolyBFTConfig) IsBridgeEnabled() bool {
 	return p.Bridge != nil
 }
 
+// Validator represents public information about validator accounts which are the part of genesis
 type Validator struct {
-	Address types.Address `json:"address"`
-	BlsKey  string        `json:"blsKey"`
-	Balance *big.Int      `json:"balance"`
-	NodeID  string        `json:"-"`
+	Address types.Address
+	BlsKey  string
+	Balance *big.Int
+	NodeID  string
 }
 
 type validatorRaw struct {
 	Address types.Address `json:"address"`
 	BlsKey  string        `json:"blsKey"`
 	Balance *string       `json:"balance"`
+	NodeID  string        `json:"nodeId"`
```

Comment:
Important point:
the same `Validator` struct is being used for both `manifest` and `genesis` files generation, therefore both of these would have p2p `NodeID` persisted (although it is not needed by the `genesis` per se, it is required in manifest, because that info is being used to populate bootnodes if those aren't provided via CLI).